sdk/core/azure_core_test/Cargo.toml

```diff
@@ -17,6 +17,8 @@ rust-version.workspace = true
 async-trait.workspace = true
 azure_core = { workspace = true, features = ["test"] }
 azure_core_test_macros.workspace = true
+azure_identity.workspace = true
```

Comment:
There's an interesting layering challenge here. This depends on azure_identity *never* taking a dependency on azure_core_test.

This has been the subject of much discussion in the C++ repo. My personal take is that the layering is azure_core->azure_identity->azure_core_test but that's not completely consistent with the directory layout.

Reply:
Loops don't exist* between dependencies and dev-dependencies, though.

*) Well, almost. @hallipr recently ran into an issue testing the publishing pipeline that seems to ignore the distinction and, while likely a bug, doesn't change the fact it's a problem in that case.

Reply:
Per offline discussion, `azure_identity` *could* have a `dev-dependencies` on `azure_core_test` though, in practice, that's hard to do since we have to sanitize everything anyway. Still, shouldn't be a problem if we end up doing that, though.

Also, I'll mark this and the associated macros crate as `publish = false` since we really don't need to publish them and, as you pointed out, Azure/azure-sdk-for-cpp doesn't either.